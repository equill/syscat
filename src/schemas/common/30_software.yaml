name: software
version: 1

resourcetypes:
  applications:
    notes: Software application of any kind.
    attributes:
      description:
        description: Comments about this particular app.
      monitor:
        description: Whether this application is to be monitored.
  operatingSystems:
    dependent: 'true'
  operatingSystemVersions:
    dependent: 'true'
    attributes:
      endoflife:
      endofsupport:

relationships:
- uri: /makes/Produces/operatingSystems
  cardinality: 1:many
  dependent: 'true'
- uri: /operatingSystems/Version/operatingSystemVersions
  cardinality: 1:many
  dependent: 'true'
- uri: /devices/OperatingSystem/operatingSystems
  cardinality: many:1
- uri: /devices/OperatingSystem/operatingSystemVersions
  cardinality: many:1
- uri: /applications/RunsOn/devices
  cardinality: many:many
- uri: /applications/ConnectsTo/applications
  cardinality: many:many
