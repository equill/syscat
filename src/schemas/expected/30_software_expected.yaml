name: software_expected
version: 1

resourcetypes:
  preferredOsVersion:
    notes: For being specific about the circumstances under which a given version is preferred
    attributes:
      description:
      mandatory:

relationships:
# Preferred OS Version
- uri: /models/PreferredOsVersion/operatingSystemVersions
  cardinality: many:1
  notes: "For the simple case where there's a global preference within this instance of Syscat"
- uri: /models/PreferredOsVersion/preferredOsVersion
  cardinality: many:1
- uri: /preferredOsVersions/Version/operatingSystemVersions
  cardinality: many:1
- uri: /preferredOsVerion/Organisation/organisations
  cardinality: many:1
- uri: /preferredOsVerion/Environment/environments
  cardinality: many:1
- uri: /preferredOsVerion/Model/models
  cardinality: many:1
- uri: /applications/DependsOn/applications
  cardinality: many:many
- uri: /applications/BusinessOwner/people
  cardinality: many:1
- uri: /applications/BusinessOwner/organisations
  cardinality: many:1
- uri: /applications/TechnicalOwner/people
  cardinality: many:1
- uri: /applications/TechnicalOwner/organisations
  cardinality: many:1
#
# Monitoring
#
- uri: /applications/Monitor/applications
  cardinality: many:many
- uri: /applications/Monitor/devices
  cardinality: many:many
- uri: /applications/Monitor/networkInterfaces
  cardinality: many:many
- uri: /applications/Monitor/bgp4Peerings
  cardinality: many:many
